watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
+            "precipitation-point",
+            "corrosion-rates",
+        ]:
+            raise RuntimeError(
+                f"Failed to configure Flash. Invalid method: {flash_method}"
+            )
 
-        # build entries for temperature and pressure
-        self.water_analysis_properties["Temperature"].update(
-            {"value": state_vars["temperature"]}
-        )
-        self.water_analysis_input_list.append(
-            self.water_analysis_properties["Temperature"]
-        )
-        self.water_analysis_properties["Pressure"].update(
-            {"value": state_vars["pressure"]}
-        )
-        self.water_analysis_input_list.append(
-            self.water_analysis_properties["Pressure"]
-        )
+        input_dict = {}
 
-        for component in state_vars["components"]:
-            charge = get_charge(component)
-            name = get_oli_name(component)
-            conc = self._oli_units(
-                state_vars["components"][component], state_vars["units"]["components"]
-            )
+        if not inflows:
+            raise RuntimeError("Inflows must be defined for Flash Analysis.")
+
+        temp_input = {
+            "unit": self.input_unit_set["temperature"]["oli_unit"],
+            "value": 273.15,
+        }
+        if temperature:
+            if float(temperature):
+                temp_input.update({"value": float(temperature)})
+            else:
+                raise RuntimeError(
+                    f"Invalid temperature: {temperature}. Expected number"
+                )
+        input_dict["temperature"] = temp_input
+
+        pres_input = {
+            "unit": self.input_unit_set["pressure"]["oli_unit"],
+            "value": 101325,
+        }
+        if pressure:
+            if float(pressure):
+                pres_input.update({"value": float(pressure)})
+            else:
+                raise RuntimeError(f"Invalid pressure: {pressure}. Expected number")
+        input_dict["pressure"] = pres_input
+
+        if flash_method in [
+            "bubblepoint",
+            "dewpoint",
+            "vapor-amount",
+            "vapor-fraction",
+            "isochoric",
+        ]:
+            if calculated_variable:
+                _logger.info(
+                    f"{flash_method} will calculate {calculated_variable} as its variable"
+                )
+                input_dict["calculatedVariable"] = calculated_variable
+            else:
+                raise RuntimeError(
+                    f"Missing argument for {flash_method}: 'calculated_variable'"
+                )
+
+        if flash_method == "isenthalpic":
+            enth_input = {
+                "unit": self.input_unit_set["enthalpy"]["oli_unit"],
+                "value": None,
+            }
+            if float(enthalpy):
+                enth_input.update({"value": float(enthalpy)})
+            else:
+                raise RuntimeError(f"Invalid enthalpy: {enthalpy}. Expected number")
```

Comment:
ValueError? Same applies for all exceptions related to case where number was expected but not provided